### ISSUE TITLE
2022.1 distro bombs

### ISSUE THREAD

**Issue by RobertHarper** [smlnj/smlnj#261]
### Version

2021.1

### Operating System

- [ ] Any
- [ ] Linux
- [X] macOS
- [ ] Windows
- [ ] Other Unix

### OS Version

Ventura 13.1

### Processor

- [ ] Any
- [ ] Arm (using Rosetta)
- [X] x86-64 (64-bit)
- [ ] Other

### System Component

Installation

### Severity

Major

### Description

Starts, but then seg faults on anything.

### Transcript

> sml
Standard ML of New Jersey [Version 2022.1; 64-bit; August 25, 2022]
- val x =3;
zsh: segmentation fault  sml

### Expected Behavior

I'm old-fashioned, but I guess I was expecting it to respond

val x : int = 3

and accept more input.

### Steps to Reproduce

Install and do what I said.

### Email address

rwh@cs.cmu.edu

**Comment by HarrisonGrodin** [comment]
I get the same thing on @ProjectSavanna/setup-smackage#63 on macOS 11; it seems that 2022.1 only works on macOS 12. Here's the output trace from GitHub Actions trying to run the installation script `go-nj.sml` on macOS 11:
```
sml src/go-nj.sml
dyld: Symbol not found: __ZTTNSt3__114basic_ifstreamIcNS_11char_traitsIcEEEE
  Referenced from: /usr/local/smlnj/bin/.run/run.amd64-darwin (which was built for Mac OS X 12.0)
  Expected in: /usr/lib/libc++.1.dylib
 in /usr/local/smlnj/bin/.run/run.amd64-darwin
```
(The same thing succeeded on macOS 12, the `macos-latest` job.)

**Comment by JohnReppy** [comment]
This issue is fixed in the 2023.1 version.
